src/validation.cpp

```diff
@@ -365,18 +377,13 @@ void CChainState::MaybeUpdateMempoolForReorg(
         LockPoints lp = it->GetLockPoints();
         const bool validLP{TestLockPointValidity(m_chain, lp)};
         CCoinsViewMemPool view_mempool(&CoinsTip(), *m_mempool);
-        // CheckSequenceLocks checks if the transaction will be final in the next block to be
-        // created on top of the new chain. We use useExistingLockPoints=false so that, instead of
-        // using the information in lp (which might now refer to a block that no longer exists in
-        // the chain), it will update lp to contain LockPoints relevant to the new chain.
-        if (!CheckSequenceLocks(m_chain.Tip(), view_mempool, tx, flags, &lp, validLP)) {
-            // If CheckSequenceLocks fails, remove the tx and don't depend on the LockPoints.
-            return true;
-        } else if (!validLP) {
-            // If CheckSequenceLocks succeeded, it also updated the LockPoints.
-            // Now update the mempool entry lockpoints as well.
+        if (!validLP) {
```

Comment:
Note that we do have a bit of a behavior change here: we now update the mempool entry LockPoints even if we're going to remove the entry later. I'm not sure if this is desirable.

I suggest adding documentation as well, e.g. "If the cached LockPoints are now invalid, recalculate them and update the mempool entry before calling `CheckSequenceLocks()`.